src/utils/auth_utils.js

```diff
@@ -42,11 +42,113 @@ const NPOBackend = axios.create({
   withCredentials: true,
 });
 
+// Sets a cookie in the browser
+const setCookie = (key, value, config) => {
+  let cookie = `${key}=${value}; max-age=${config.maxAge}; path=${config.path}`;
+
+  if (config.domain) {
+    cookie += `; domain=${config.domain}`;
+  }
+  if (cookie.secure) {
+    cookie += '; secure';
+  }
+  document.cookie = cookie;
+};
+
+/**
+ * Returns the current user synchronously
+ * @param {Auth} authInstance
+ * @returns The current user (or undefined)
+ */
+const getCurrentUser = authInstance =>
+  new Promise((resolve, reject) => {
+    const unsubscribe = authInstance.onAuthStateChanged(user => {
+      unsubscribe();
+      resolve(user);
+    }, reject);
```

Comment:
please return error if rejecting, https://stackoverflow.com/questions/63047603/attempting-to-make-firebase-auth-currentuser-a-promise/63064662#63064662
TLDR, change to be 
```
err => {
    reject(err)
  })
```